Change ownership of the EFS share mount point

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This PR adjusts the cloud-config logic for setting up the EFS share to `chown` the mount point using the provided owner and group.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Currently the mount point is created as the `root` user. This means you must have `root` privileges to access the share. On Kali machines this is fine because `sudo` permissions are configured, but since Windows instances must go through a Samba instance this creates a problem. Currently a Windows instance cannot write to the EFS share until a sub-directory with the appropriate `uid` (the `vnc` and `smbguest` `uid`) has been created. With this in place the Windows instance will be able to immediately write to the EFS share and other system users will not need to `sudo` to access the share.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
Automated tests pass. I deployed this in my testing environment and confirmed that the share is being `chown`ed correctly and the Windows instance is able to write to the root EFS directory with no additional changes.
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

## ✅ Checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Modify EFS mount cloud-config

Add a command to `chown` the EFS mount point using provided owner and group
names. These should correspond to an existing user and group on the instance.

dav3r commented on assessorportal_cloud_init.tf:
```diff
@@ -18,7 +18,9 @@ data "cloudinit_config" "assessorportal_cloud_init_tasks" {
         efs_id      = aws_efs_mount_target.target[var.private_subnet_cidr_blocks[0]].file_system_id
+        group       = "vnc"
         mount_point = "/share"
+        owner       = "vnc"
```
Not to complicate things, but we have tried to avoid hard-coding the username as `vnc`.  Instead, we should obtain the username by looking up the value of the SSM parameter specified by `var.ssm_key_vnc_username`.  I'm fine with you making a TODO issue to fix this later if you don't have time for it now.

For an example of how we do this, take a look at [`cloud-init/write-aws-config.tpl.sh`](https://github.com/cisagov/cool-assessment-terraform/blob/develop/cloud-init/write-aws-config.tpl.sh).

As long as we are doing this, it might also be a good idea to add the VNC group to SSM as well, though I'm fine assuming the username and group are the same for now.

mcdonnnj replied:
The only issue I see is that this is a cloud-config file and not a shell script. Thoughts on using an [`aws_ssm_parameter` data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssm_parameter)?

dav3r replied:
I don't love this bit with the `aws_ssm_parameter` data source:
```
The unencrypted value of a SecureString will be stored in the raw [Terraform] state as plain-text.
```

In this case, it wouldn't be that hard to turn this cloud-config file into a shell script.  Or maybe just split out the `runcmd` bits into their own shell script if we don't want to lose the simplicity of using that `mounts` command.

dav3r replied:
PS: I'm not worried about this particular SSM value being in the Terraform state- just more generally about us going down that path and ending up with something more sensitive landing in the state.

mcdonnnj replied:
Yeah I only suggested it given the non-sensitive nature of this value. We could always add a disclaimer that the value is non-sensitive so it's fine if it's in the Terraform state?

dav3r replied:
I suppose that's ok in this case.  Though if it's not sensitive, then maybe it doesn't even belong in SSM?  Though it is convenient to have certain values in SSM if they are accessed by Terraform, cloud-init, and packer scripts.

jsf9k replied:
> Though if it's not sensitive, then maybe it doesn't even belong in SSM?

SSM Parameter Store is for storing parameters, secret or otherwise.  There's nothing wrong with storing non-secret parameters there for convenience, although in many cases it may be simpler to store such parameters in, e.g. a `*.tfvars` file.

mcdonnnj replied:
Please enjoy d0f2f37 for a non-static version.

dav3r replied:
Perfecto - nice work.

dav3r reviewed (changes requested):
I've got one thang I'd like to see addressed before I approve.

mcdonnnj committed:
Use SSM Parameter Store parameters

Replace the statically declared values for the owner/group used in the
efs-mount cloud-config template file with values from SSM Parameter Store
parameters. Since these values are used when building the AMIs used in this
Terraform configuration it will allow this configuration to use the correct
values when it is applied.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj committed:
Update variable description

Update the `ssm_key_vnc_username` variable's description to make it consistent
with similar variables.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
Looks fantastic now! 🥇
